Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm

```diff
@@ -349,6 +349,25 @@
 
     [screenTimeWebpageController setURL:[webView _mainFrameURL]];
 }
+
+void PageClientImplCocoa::setScreenTimePictureInPicture(bool value)
+{
+    RetainPtr screenTimeWebpageController = [webView() _screenTimeWebpageController];
+    if (!screenTimeWebpageController)
+        return;
+
+    [screenTimeWebpageController setURLIsPictureInPicture:value];
+}
+
+void PageClientImplCocoa::setScreenTimeIsPlayingVideo(bool value)
+{
+    RetainPtr screenTimeWebpageController = [webView() _screenTimeWebpageController];
+    if (!screenTimeWebpageController)
+        return;
+
+    [screenTimeWebpageController setURLIsPlayingVideo:value];
+}
```

Comment:
Do we need to skip setting this for non-persistent datastores? Or is that already handled by not donating the URL itself in that scenario.